### ISSUE TITLE
[TG-nonlin] Segmentation fault during parse(loadFPfromFile)

### ISSUE THREAD

**Issue by izlatkin** [izlatkin/solidity_testgen#13]
```
tgnonlin --inv-mode 0 --no-term --keys 4271,13242 ~/Downloads/testgen_output/loops/while_loop_array_assignment_memory_memory/while_loop_array_assignment_memory_memory_wo_adt.smt2 
(error "line 18 column 9448: array operation requires one sort parameter (the array sort)")
Segmentation fault: 11

```
[while_loop_array_assignment_memory_memory_wo_adt.smt2.txt](https://github.com/izlatkin/solidity_testgen/files/9041640/while_loop_array_assignment_memory_memory_wo_adt.smt2.txt)
[while_loop_array_assignment_memory_memory_updated.sol.txt](https://github.com/izlatkin/solidity_testgen/files/9041641/while_loop_array_assignment_memory_memory_updated.sol.txt)
![image](https://user-images.githubusercontent.com/16242682/177206785-d9062c1e-71ad-42f4-90f9-13ce52b14de5.png)

158 benchmarks failing because of the same error

**Comment by grigoryfedyukovich** [comment]
It's something with Leo's conversion. The formula has this expression:
`(= a_45_length_pair_1_uint_array_tuple_accessor_array_25 (const 0))`
And the parser complains about `(const 0)`.
This bug is reproducible even if you call pure Z3.

**Comment by izlatkin** [comment]
same with: 
```/Users/ilyazlatkin/CLionProjects/aeval/cmake-build-debug/tools/nonlin/tgnonlin --keys contract_C:^check: ../sandbox/simple_if_array_wo_adt.smt2
Segmentation fault: 11

>z3 simple_if_array_wo_adt.smt2
(error "line 63 column 4004: array operation requires one sort parameter (the array sort)")
sat
```
but
``` 
z3 simple_if_array.smt2 
unsat
```

**Comment by leonardoalt** [comment]
`sed -i 's/(const 0)/((as const (Array Int Int)) 0)/' file`

**Comment by izlatkin** [comment]
@leonardoalt  thanks.  Issue fixed: c9540ba9b17b95642a106fb0fa487f56a7c070c6
